### PR TITLE
Fixes Deferred.wait(_:) when filled

### PR DESCRIPTION
**Pull request by zwaldowski**
Addresses #67.

### CODE REVIEW

File: Sources/Deferred.swift [hunk 1]
```diff
@@ -91,18 +91,26 @@ public struct Deferred<Value>: FutureType, PromiseType {
...
-        let queue = dispatch_get_global_queue(QOS_CLASS_UTILITY, 0)
-        var value: Value?
-        let handler = upon(queue, options: DISPATCH_BLOCK_ENFORCE_QOS_CLASS) {
-            value = $0
+        var result: Value?
+        func assign(value: Value) {
+            result = value
+        }
+
+        // Pure FutureType implementations implement isFilled in terms of wait()
```
Comment (jgallagher) [inline Sources/Deferred.swift]: This comment implies the next line would be infinitely circular, wouldn't it? If wait calls isFilled and isFilled is implemented in terms of wait.

File: Sources/Deferred.swift [hunk 2]
```diff
@@ -91,18 +91,26 @@ public struct Deferred<Value>: FutureType, PromiseType {
...
+        if case .Now = time where isFilled {
```
Comment (jgallagher) [inline Sources/Deferred.swift]: Why only do this for `.Now` and not all timeouts?

Comment (zwaldowski) [inline Sources/Deferred.swift]: …fair.

File: Sources/Deferred.swift (hunk 1 above)
Comment (zwaldowski) [inline Sources/Deferred.swift]: No, we implement `isFilled` ourselves. "Pure" meant as "Only" (i.e., not `PromiseType`). Will clarify.

**Comment by zwaldowski** [comment]
Comments addressed.

**Comment by jgallagher** [comment]
:+1:

**Pull request merged**